contracts/sg721-updatable/src/contract.rs

```diff
@@ -147,40 +150,59 @@ pub fn execute_update_royalty_info(
     let payment_addr = deps.api.addr_validate(&payment_address)?;
     let share = Decimal::percent(share_bps) / Uint128::from(100u128);
     let max_share_limit = Decimal::percent(MAX_ROYALTY_SHARE_BPS) / Uint128::from(100u128);
-
-    if share > max_share_limit {
-        return Err(ContractError::RoyaltyShareIncreasedTooMuch {});
-    }
-
-    let current_royalty_info = collection_info.royalty_info;
-
-    if let Some(curr_royalty_info) = current_royalty_info {
-        let updated_at = match curr_royalty_info.updated_at {
-            Some(updated_at) => updated_at,
-            // If current royalty info updated_at is None, it might be an older version that has not been migrated properly.
-            None => return Err(ContractError::RoyaltyInfoInvalid {}),
-        };
-
-        // make sure the update time is after 24 hours
+    let max_share_delta = Decimal::percent(MAX_ROYALTY_SHARE_DELTA_BPS) / Uint128::from(100u128);
+    ensure!(
+        share <= max_share_limit,
+        // multiply by 100 to get the percentage for error message
+        ContractError::InvalidRoyalties(format!(
+            "Share percentage cannot be greater than {}%",
+            max_share_limit * Uint128::from(100u128)
+        ))
+    );
+
+    let updated_at = ROYALTY_UPDATED_AT.load(deps.storage)?;
+
+    if let Some(old_royalty_info) = collection_info.royalty_info {
+        // make sure the update time is at least 24 hours after the last one
         let end = (Expiration::AtTime(updated_at) + DAY)?;
-        let update_too_soon = !end.is_expired(&env.block);
-
-        if share > curr_royalty_info.share && update_too_soon {
-            return Err(ContractError::RoyaltyUpdateTooSoon {});
+        ensure!(
+            end.is_expired(&env.block),
+            ContractError::RoyaltyUpdateTooSoon {}
+        );
+        // Make sure the share is not increased more than MAX_ROYALTY_SHARE_DELTA_BPS at a time
+        if share > old_royalty_info.share {
+            let share_delta = share - old_royalty_info.share;
```

Comment:
what if `share < old_royalty_info.share`

Reply:
The reasoning was that the upwards updates are the risky ones that may cause traders to lose money if the change is too steep.